pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaMPI.hpp

```diff
@@ -401,9 +401,9 @@ class StateVectorCudaMPI final
         } else if (opName == "Rot" || opName == "CRot") {
             auto rot_matrix =
                 adjoint
-                    ? cuGates::getRot<CFP_t>(params[2], params[1], params[0])
+                    ? cuGates::getRot<CFP_t>(-params[2], -params[1], -params[0])
                     : cuGates::getRot<CFP_t>(params[0], params[1], params[2]);
-            applyDeviceMatrixGate(rot_matrix.data(), ctrls, tgts, adjoint);
+            applyDeviceMatrixGate(rot_matrix.data(), ctrls, tgts, false);
```

Comment:
it seems a bit odd to always pass through `adjoint=false` in  `applyDeviceMatrixGate`! But I guess there's no better way if we want to pass a reversed rot matrix to this method. Could we fix this by adding the matrix evaluation (line 402) within `applyDeviceMatrixGate` instead?

Reply:
We could keep `rot_matrix` to be the one with adjoint=false, and let `applyDeviceMatrixGate` handle the adjoint (is this what you mean?), but I believe that would be less efficient than passing a reversed rot matrix directly. Thoughts @multiphaseCFD?

Reply:
I'm fine with both approaches. I'll leave you guys to decide. Please add a note if this hard-coded `false` is adopted. Is it possible to change to signature of `getRot` from `getRot(param0, param1, param2)` to (`getRot(params, adjoint)`)?

Reply:
This implementation is similar to e.g. what's been used in[ LQ ](https://github.com/PennyLaneAI/pennylane-lightning/blob/45a67d495552ca9aca3c05467b16b1b6d2e1d4ab/pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp#L1141), so to keep it consistent I will use this. Actually `applyDeviceMatrixGate` also has `use_adjoint=false`, so I will remove the argument, and add a note.

Reply:
I'm okay with keeping this as is for the sake of consistency in this PR! However, I’m still not persuaded by [the reasoning behind this](https://github.com/PennyLaneAI/pennylane-lightning/pull/1004#discussion_r1856959574).

Reply:
Just did some test and indeed the performance is the same, happy to let custatevec api deal with the adjoint, which does give a cleaner approach.